tests/test_enum.rs

```diff
@@ -13,6 +13,13 @@ pub enum MyEnum {
     OtherVariant,
 }
 
+#[pyclass(ord)]
+#[derive(Debug, PartialEq, Eq, Clone)]
```

Comment:
A thought: should we require a `PartialOrd` implementation for `pyclass(ord)`? I see two upsides:
- That would allow us to skip casting via `__pyo3__int__`.
- I think for structs and other data classes we would need to use the `PartialOrd` implementation anyway as `__pyo3__int__` wouldn't be possible, so again this seems good for consistency.

If we agree, as far as implementation would go, just remove the calls to convert with `__pyo3__int__` from these branches.

Reply:
That was also one of my first thoughts, but then we should probably also require `PartialEq` for the already existing impls.

Reply:
Requiring PartialOrd seems reasonable to me.  Would requiring PartialEq break anything for current implementations?  If we require PartialOrd, would the ensure_spanned checks still be required?

Reply:
I think requiring `PartialEq` would break compiles, but it'd also be trivial to fix for these simple enum cases (just add the derive). @Icxolu would you be ok if we agree that we should require `PartialEq` but postpone changing that to a future PR / tracking issue where we can also add `#[pyclass(eq)]` at the same time?

I also would love to have `#[pyclass(hash)]` to be implemented (again, requiring `Hash`).

... and then a future step is working out how to apply all of these to structs to get dataclass-like functionality :)

Reply:
> If we require PartialOrd, would the ensure_spanned checks still be required?

If you mean the checks for `ord` not being accepted on structs, we'll still need those (unless we extend this PR to make `ord` work for every type of `#[pyclass]`, which I think should be possible with a bit more work).

Reply:
> If you mean the checks for `ord` not being accepted on structs, we'll still need those (unless we extend this PR to make `ord` work for every type of `#[pyclass]`, which I think should be possible with a bit more work).

Yes,  I think making `ord` work for all of these types would be great in this PR.  I will scope this out and raise any questions along the way.